src/utils/signer.rs

```diff
@@ -0,0 +1,161 @@
+use crate::utils::from_env::FromEnv;
+use alloy::{
+    consensus::SignableTransaction,
+    primitives::{Address, ChainId, B256},
+    signers::{
+        aws::{AwsSigner, AwsSignerError},
+        local::{LocalSignerError, PrivateKeySigner},
+        Signature,
+    },
+};
+use aws_config::{load_defaults, BehaviorVersion};
+use aws_sdk_kms::Client;
+use std::borrow::Cow;
+
+/// Configuration for a LocalOrAws signer.
+///
+/// Usage:
+/// ```
```

Comment:
this will cause a doc compile error unless you add all appropriate `use` statements, and generally make it a runnable test.

see here for docs:
https://doc.rust-lang.org/rustdoc/write-documentation/documentation-tests.html

and here for a real-world example
https://github.com/init4tech/trevm/blob/main/src/lib.rs#L40-L54 

https://doc.rust-lang.org/rustdoc/write-documentation/documentation-tests.html

Reply:
oh, oop 🤔  wait, why did this compile then?

Reply:
the block is treated as an entire separate binary, so you need to import using the crate name

Reply:
i got it i got it :)